MergerService/src/run.cs

```diff
@@ -0,0 +1,166 @@
+                Thread.Sleep(500);
+
+                MergeTask? task = MergeTask.GetTask();
+
+                // Guard clause in case there are no batches or sources
+                if (task == null || task.Batches == null || task.Sources == null)
+                {
+                    continue;
+                }
+
+                foreach (Bounds bounds in task.Batches)
+                {
+                    stopWatch.Reset();
+                    stopWatch.Start();
+
+                    int totalTileCount = bounds.Size();
+                    int tileProgressCount = 0;
+
+                    List<Data> sources = BuildDataList(task.Sources, totalTileCount);
+                    Data target = sources[0];
+
+                    List<Tile> tiles = new List<Tile>(totalTileCount);
+
+                    Console.WriteLine($"Total amount of tiles to merge: {totalTileCount}");
+
+                    // Go over the bounds of the current batch
+                    for (int x = bounds.MinX; x < bounds.MaxX; x++)
+                    {
+                        for (int y = bounds.MinY; y < bounds.MaxY; y++)
+                        {
+                            Coord coord = new Coord(bounds.Zoom, x, y);
+
+                            // Create tile builder list for current coord for all sources
+                            List<CorrespondingTileBuilder> correspondingTileBuilders = new List<CorrespondingTileBuilder>();
+                            foreach (Data source in sources)
+                            {
+                                correspondingTileBuilders.Add(() => source.GetCorrespondingTile(coord, true));
+                            }
+
+                            byte[]? blob = Merge.MergeTiles(correspondingTileBuilders, coord);
+
+                            if (blob != null)
+                            {
+                                tiles.Add(new Tile(coord, blob));
+                            }
+
+                            tileProgressCount++;
+
+                            // Show progress every 1000 tiles
+                            if (tileProgressCount % 1000 == 0)
+                            {
+                                Console.WriteLine($"Tile Count: {tileProgressCount} / {totalTileCount}");
+                            }
+                        }
+                    }
+
+                    target.UpdateTiles(tiles);
+
+                    foreach (Data data in sources)
+                    {
+                        target.UpdateMetadata(data);
+                    }
+
+                    Console.WriteLine($"Tile Count: {tileProgressCount} / {totalTileCount}");
+                    target.Wrapup();
+
+                    stopWatch.Stop();
+
+                    // Get the elapsed time as a TimeSpan value.
+                    ts = stopWatch.Elapsed;
+                    Console.WriteLine("Merged the following bounds:");
```

Comment:
this log should be also before the actual merging (with the Total amount of tiles to merge log)